### ISSUE TITLE
 The training issue

### ISSUE THREAD

**Issue by mayang113** [yzshi5/GM-GANO#7]
![屏幕截图 2024-04-26 124526](https://github.com/yzshi5/GM-GANO/assets/166595329/70a13e56-455f-45c1-b1b1-fb663e51938d)
![屏幕截图 2024-04-26 124619](https://github.com/yzshi5/GM-GANO/assets/166595329/5a6f923a-4dd6-4af8-b349-e607a682deef)
Hello, I would like to ask if this training process is normal? How good is the value of loss_G to represent good results?

**Comment by yzshi5** [comment]
From our experience, at the beginning of training, we may encounter abnormal loss, but the loss converges along the training. That is not a problem.

Besides, I updated the code of `GANO_model.py`, you can use the kernel functional (instead of mean) in the last layer of discriminator, the loss will be more stable. However, we didn't observe the much improvement of the performance with the changed functional.

![image](https://github.com/yzshi5/GM-GANO/assets/109268435/8568078f-7e25-4c57-b198-6a6fa3869ca6)

**Comment by mayang113** [comment]
![屏幕截图 2024-04-26 134942](https://github.com/yzshi5/GM-GANO/assets/166595329/d6aeab62-b05b-40c9-9512-8512e81e6087)
Thank you for your response. Is the current value reasonable? I have 8100 time series data with a batch size of 64.

**Comment by yzshi5** [comment]
It's hard to say whether the loss value is reasonable or not, since it can be quite different for different datasets. The loss of GANO is not well explained, (claimed by the authors of GANO), but it can indeed learn the mapping. I would suggest not caring too much about the loss. Just check the loss for the last few epochs.

**Comment by mayang113** [comment]
![屏幕截图 2024-04-26 140539](https://github.com/yzshi5/GM-GANO/assets/166595329/4a90247b-9071-403b-8c2a-2a0f8d79531e)
Ok thanks for the reply this is my last question , this is my attr_file I just need to make sure that the order of the rows matches the order in the data_file? Do the columns need to be in order? Also does the name of this seismic event need to match the data_file?

**Comment by yzshi5** [comment]
Yes, order of rows must match the order of data. That's is the only requirement.
